src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -230,71 +231,36 @@ private bool TryBindUnconvertedBinaryOperatorToDefaultInterpolatedStringHandler(
                 return false;
             }
 
-            bool isConstant = true;
-            var stack = ArrayBuilder<BoundBinaryOperator>.GetInstance();
+            // The constant value is folded as part of creating the unconverted operator. If there is a constant value, then the top-level binary operator
+            // will have one.
+            bool isConstant = binaryOperator.ConstantValue is not null;
             var partsArrayBuilder = ArrayBuilder<ImmutableArray<BoundExpression>>.GetInstance();
-            int partsCount = 0;
 
-            BoundBinaryOperator? current = binaryOperator;
-
-            while (current != null)
-            {
-                Debug.Assert(current.IsUnconvertedInterpolatedStringAddition);
-                stack.Push(current);
-                isConstant = isConstant && current.Right.ConstantValue is not null;
-                var rightInterpolatedString = (BoundUnconvertedInterpolatedString)current.Right;
-
-                if (!ValidateInterpolatedStringParts(rightInterpolatedString))
+            if (!binaryOperator.VisitBinaryOperatorInterpolatedString(partsArrayBuilder, static (BoundUnconvertedInterpolatedString unconvertedInterpolatedString, ArrayBuilder<ImmutableArray<BoundExpression>> partsArrayBuilder) =>
```

Comment:
>static (BoundUnconvertedInterpolatedString unconvertedInterpolatedString, ArrayBuilder<ImmutableArray<BoundExpression>> partsArrayBuilder) => [](http://example.com/codeflow?start=89&length=141)

Consider placing the lambda header on a separate line. This should make it easier to figure out what the block on the next line is about. Alternatively the body can be aligned with the header.